packages/nimble/R/MCMC_samplers.R

```diff
@@ -548,13 +570,23 @@ sampler_ess <- nimbleFunction(
         theta_max <- theta
         model[[target]] <<- f*cos(theta) + nu*sin(theta) + target_mean
         lp <- calculate(model, calcNodes)
-        while(is.nan(lp) | lp < u) {   # must be is.nan()
+        numContractions <- 0
+        while((is.nan(lp) | lp < u) & theta_max - theta_min > eps & numContractions < maxContractions) {   # must be is.nan()
+            ## The checks for theta_max - theta_min small and max number of contractions are
+            ## for cases where model is in invalid state and lp calculations are NA/NaN or where
+            ## theta interval contracts to zero
             if(theta < 0)   theta_min <- theta   else   theta_max <- theta
             theta <- runif(1, theta_min, theta_max)
             model[[target]] <<- f*cos(theta) + nu*sin(theta) + target_mean
             lp <- calculate(model, calcNodes)
+            numContractions <- numContractions + 1
         }
-        nimCopy(from = model, to = mvSaved, row = 1, nodes = calcNodes, logProb = TRUE)
+        if(theta_max - theta_min <= eps | numContractions == maxContractions) {
```

Comment:
@paciorek Why did you use a relative difference check in `sampler_slice` but an absolute difference here?